src/yaml-support/tkn-code-actions.ts

```diff
@@ -97,6 +101,129 @@ class PipelineCodeActionProvider implements vscode.CodeActionProvider {
+    action.taskRefEndPosition = endPos;
+    action.taskRefName = taskRefName;
+    action.taskKind = selectedElement.taskRef?.kind.value;
+    action.documentUri = document.uri;
+
+    return action;
+  }
+
+  private async resolveInlineAction(codeAction: InlineTaskAction): Promise<InlineTaskAction> {
+    return vscode.window.withProgress({location: vscode.ProgressLocation.Notification, cancellable: false, title: `Loading '${codeAction.taskRefName}' Task...` }, async (): Promise<vscode.CodeAction> => {
+      const uri = tektonFSUri(codeAction.taskKind === TektonYamlType.ClusterTask ? ContextType.CLUSTERTASK : ContextType.TASK , codeAction.taskRefName, 'yaml');
+      try {
+        const taskDoc = await tektonVfsProvider.loadTektonDocument(uri, false);
+        codeAction.edit = new vscode.WorkspaceEdit();
+        codeAction.edit.replace(codeAction.documentUri,
+          new vscode.Range(codeAction.taskRefStartPosition, codeAction.taskRefEndPosition), 
+          this.extractTaskDef(taskDoc, codeAction.taskRefStartPosition.character, codeAction.taskRefEndPosition.character));
+      } catch (err){
+        vscode.window.showErrorMessage('Cannot get Tekton Task definition: ' + err.toString());
+        telemetryLogError('resolveCodeAction', `Cannot get '${codeAction.taskRefName}' Task definition`);
+      }
+      return codeAction;
+    });
+  }
+
+  private getExtractTaskAction(selectedElement: PipelineTask, document: vscode.TextDocument): ExtractTaskAction | undefined {
+    const taskSpec = selectedElement.taskSpec;
+    if (!taskSpec) {
+      return;
+    }
+    const startPos = document.positionAt(taskSpec.keyNode?.startPosition);
+    let taskSpecStartPos = document.positionAt(taskSpec.startPosition);
+    // start replace from stat of the line
+    taskSpecStartPos = document.lineAt(taskSpecStartPos.line).range.start;
+    let endPos = document.positionAt(taskSpec.endPosition);
+
+    // if last line is contains only spaces then replace til previous line
+    const lastLine = document.getText(new vscode.Range(endPos.line, 0, endPos.line, endPos.character));
+    if (lastLine.trim().length === 0) {
+      endPos = document.lineAt(endPos.line - 1).range.end;
+    }
+
+    const action: ExtractTaskAction = new vscode.CodeAction(`Extract '${selectedElement.name.value}' Task spec`, EXTRACT_TASK);
+    action.documentUri = document.uri;
+    action.taskSpecStartPosition = startPos;
+    action.taskSpecEndPosition = endPos;
+    action.taskSpecText = document.getText(new vscode.Range(taskSpecStartPos, endPos));
+
+    return action;
+  }
+
+  private async resolveExtractTaskAction(action: ExtractTaskAction): Promise<vscode.CodeAction> {
+    return vscode.window.withProgress({location: vscode.ProgressLocation.Notification, cancellable: false, title: 'Extracting Task...' }, async (): Promise<vscode.CodeAction> => {
+      try {
+        const name = await vscode.window.showInputBox({ignoreFocusOut: false, prompt: 'Provide Task Name' });
```

Comment:
I think ignoreFocusOut should be true showInputBox and  showQuickPick  close when clicking anywhere else on screen

Reply:
Yes, good catch!